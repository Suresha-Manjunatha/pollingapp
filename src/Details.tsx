/** @format */

import React, { FC, useEffect } from "react";
import { View, Text } from "react-native";
import { ScreenStyles } from "./styles";

type route = {
  route: {
    params: {
      item: any;
    };
  };
};

const Details: FC<route> = (props) => {
  useEffect(() => {
    window.scrollTo(0, 0);
  }, []);
  return (
    <View testID="details" style={ScreenStyles.screen}>
      {/* <Text style={{ fontSize: 24, color: "blue" }}>RAW JSON</Text> */}
      <Text> {props.route.params.item}</Text>
    </View>
  );
};

export default Details;
