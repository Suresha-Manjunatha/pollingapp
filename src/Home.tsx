/** @format */

import { useNavigation } from "@react-navigation/native";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import axios from "axios";
import React, { useEffect, useState } from "react";
import { FlatList, View, Text, TouchableOpacity } from "react-native";
import { ScreenStyles } from "./styles";

const Home = () => {
  const [data, setData] = useState<any[]>([]);
  const [page, setPage] = useState<number>(1);
  const navigation = useNavigation<NativeStackNavigationProp<any>>();
  let interval: NodeJS.Timer;
  const [loading, setloading] = useState<boolean>(false);

  const handelPage = () => {
    setPage((p: any) => p + 1);
  };

  const fetchData = async () => {
    try {
      setloading(true);
      const res = await axios.get(
        `https://hn.algolia.com/api/v1/search_by_date?query=story&page=${page}`
      );
      //   console.log(res.data.hits);
      setData([...data, ...res.data.hits]);
    } catch (e) {
      console.log("error");
    } finally {
      setloading(false);
    }
  };

  useEffect(() => {
    fetchData();
    console.log(page);
  }, [page]);

  useEffect(() => {
    interval = setInterval(() => handelPage(), 10000);
    return () => clearInterval(interval);
  }, []);

  const handleScroll = () => {
    if (
      window.scrollY + window.innerHeight >=
      document.documentElement.scrollHeight
    ) {
      !loading && handelPage();
    }
  };

  useEffect(() => {
    window.addEventListener("scroll", handleScroll);

    return () => window.removeEventListener("scroll", handleScroll);
  }, []);

  return (
    <View testID="home" style={ScreenStyles.screen}>
      <FlatList
        horizontal={false}
        testID="home-list"
        data={data}
        // keyExractor={(item:any)=>item.objectID}
        renderItem={({ item, index }) => {
          return (
            <TouchableOpacity
              testID={`test-${item.index}`}
              style={ScreenStyles.item}
              key={item.objectID}
              onPress={() => {
                navigation.navigate("RAW JSON", {
                  item: JSON.stringify(item, null, 4),
                });
              }}
            >
              <Text style={ScreenStyles.itemText1}>
                <Text style={ScreenStyles.title}>Title: </Text>
                {item.title ? (
                  item.title
                ) : item.story_title ? (
                  item.story_title
                ) : (
                  <i>data not found</i>
                )}
              </Text>
              <Text style={ScreenStyles.itemText1}>
                <Text style={ScreenStyles.title}>URL: </Text>
                {item.url ? (
                  <a href={item.url}>{item.url}</a>
                ) : item.story_url ? (
                  <a href={item.story_url}>{item.story_url}</a>
                ) : (
                  <i>data not found</i>
                )}
              </Text>
              <Text style={ScreenStyles.itemText1}>
                <Text style={ScreenStyles.title}>Created_at: </Text>
                {item.created_at}
              </Text>

              <Text style={ScreenStyles.itemText1}>
                <Text style={ScreenStyles.title}>Author: </Text>
                {item.author}
              </Text>
            </TouchableOpacity>
          );
        }}
        onEndReachedThreshold={0.5}
        // onEndReached={handelPage}
        // ListFooterComponent={() => {
        //   if (loading) {
        //     return <ActivityIndicator />;
        //   } else return <View />;
        // }}
      />
    </View>
  );
};

export default Home;
