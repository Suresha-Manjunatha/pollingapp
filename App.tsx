import { NavigationContainer } from "@react-navigation/native";
import { createNativeStackNavigator } from "@react-navigation/native-stack";
import { StatusBar } from "expo-status-bar";
import { StyleSheet, Text, View } from "react-native";
import Details from "./src/Details";
import Home from "./src/Home";

const App = () => {
  const { Navigator, Screen } = createNativeStackNavigator();
  const option = {
    headerShown: false,
  };
  return (
    <NavigationContainer>
      <Navigator>
        <Screen name="Home" component={Home} options={option} />
        <Screen name="RAW JSON" component={Details} />
      </Navigator>
    </NavigationContainer>
  );
};

export default App;
const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
});
