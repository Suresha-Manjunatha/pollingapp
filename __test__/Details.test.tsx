import Home from "../src/Home";
import { render } from "@testing-library/react-native";
import Details from "../src/Details";
import App from "../App";

const route = {
  params: {
    item: "",
  },
};

describe("home component test", () => {
  jest.useFakeTimers();
  it("take snap shot", () => {
    const tree = render(<Details route={route} />);
    expect(tree).toMatchSnapshot();
  });
});
