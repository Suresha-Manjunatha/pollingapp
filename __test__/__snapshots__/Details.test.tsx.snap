// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home component test take snap shot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "marginTop": 20,
      "paddingTop": 20,
    }
  }
  testID="details"
>
  <Text>
     
    
  </Text>
</View>
`;
